File: src/main/java/com/atlaspharmacy/atlaspharmacy/schedule/domain/Examination.java
package com.atlaspharmacy.atlaspharmacy.schedule.domain;

import com.atlaspharmacy.atlaspharmacy.schedule.domain.enums.AppointmentType;
import com.atlaspharmacy.atlaspharmacy.schedule.domain.valueobjects.Period;
import com.atlaspharmacy.atlaspharmacy.users.domain.Dermatologist;
import com.atlaspharmacy.atlaspharmacy.users.domain.Patient;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;


@Entity
@DiscriminatorValue(value = AppointmentType.Values.Examination)
public class Examination extends Appointment{

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Dermatologist dermatologist;

    public Examination() {
    }

    public Examination(Dermatologist dermatologist) {
        this.dermatologist = dermatologist;
    }

    public Examination(Period appointmentPeriod, double cost, String type, boolean isCanceled, Dermatologist dermatologist, Patient patient) {
        super(appointmentPeriod, cost, type, isCanceled, patient);
        this.dermatologist = dermatologist;
    }

    public Dermatologist getDermatologist() {
        return dermatologist;
    }

    public void setDermatologist(Dermatologist dermatologist) {
        this.dermatologist = dermatologist;
    }
}

File: frontend/src/app/model/appointment/examination.ts
import {Dermatologist} from "@app/model/users/dermatologist/dermatologist";
import {Period} from "@app/model/appointment/period";
import {Patient} from "@app/model/users/patient/patient";
import {Pharmacy} from "@app/model/pharmacy/pharmacy";

export class Examination {
  public version:Number;
  public id:Number;
  public appointmentPeriod:Period;
  public cost: Number;
  public type:String;
  public isCanceled: boolean;
  public patient:Patient;
  public pharmacy:Pharmacy;
  public finished:Boolean;
  public dermatologist: Dermatologist;

  constructor(version:Number,id:Number,appointmentPeriod:Period,cost:Number,type:String,isCanceled:boolean
              ,patient:Patient,pharmacy:Pharmacy,finished:boolean,dermatologist:Dermatologist) {
    this.id=id;
    this.appointmentPeriod=appointmentPeriod;
    this.cost=cost;
    this.isCanceled=isCanceled;
    this.patient=patient;
    this.pharmacy=pharmacy;
    this.dermatologist=dermatologist;
    this.version = version;
    this.finished = finished;
    this.type=type;
  }

}
